src/mut/section.cpp

```diff
@@ -89,7 +89,7 @@ std::ostream& operator<<(std::ostream& os, const std::shared_ptr<Section>& secti
     return os;
 }
 
-std::shared_ptr<Section> Section::appendSection(const std::shared_ptr<Section>& original_section,
+std::shared_ptr<Section> Section::appendSection(std::shared_ptr<Section> original_section,
```

Comment:
Can you add this comment at line 113 (GH does not allow comments on unchanged lines...):

```
        // Careful not to use a reference here or you will face ref invalidation problem with vector resize
        // The argument `original_section` of this function could be a reference to the `_children` array 
        // and that reference might be invalidated by this `push_back` (in case `vector` needs to reallocate the array)
```